### PR TITLE
Fix #1516 send noimage.png if user is removed.

### LINKED ISSUE #1516: getting avatars of disabled accounts results in 404 (as of 4.11)
`_unknown/_avatar` returns 404 under GitBucket 4.11

returning an image is expected (gray-man is okay).

### PR DESCRIPTION
**Pull request by kounoike**
This PR fixes #1516.
Sorry, I didn't care to removed user in #1497.

Before:
205518-manage users - google chrome

After:
205802-manage users - google chrome

### Before submitting a pull-request to Gitbucket I have first:

- [x] read the [contribution guidelines](https://github.com/gitbucket/gitbucket/blob/master/.github/CONTRIBUTING.md)
- [ ] rebased my branch over master
- [x] verified that project is compiling
- [x] verified that tests are passing (mysql test is failed by timeout)
- [ ] squashed my commits as appropriate *(keep several commits if it is relevant to understand the PR)*
- [x] [marked as closed using commit message](https://help.github.com/articles/closing-issues-via-commit-messages/) all issue ID that this PR should correct

### CODE REVIEW

File: src/main/scala/gitbucket/core/controller/AccountController.scala [hunk 1]
```diff
@@ -150,20 +150,21 @@ trait AccountControllerBase extends AccountManagementControllerBase {
...
       account.image.map{ image =>
         RawData(FileUtil.getMimeType(image), new java.io.File(getUserUploadDir(userName), image))
       }.getOrElse{
-        contentType = "image/png"
         (if (account.isGroupAccount) {
           TextAvatarUtil.textGroupAvatar(account.fullName)
         } else {
           TextAvatarUtil.textAvatar(account.fullName)
-        }).getOrElse(Thread.currentThread.getContextClassLoader.getResourceAsStream("noimage.png"))
+        }).get
```
Comment (takezoe) [inline src/main/scala/gitbucket/core/controller/AccountController.scala]: Don't use `Option.get` if there are any possibility to `None` is got here. If `TextAvatarUtil` can always return an image, it shouldn't be `Option`.
I think that to fallback to `noimage.png` is good. Why did you remove that?

Comment (kounoike) [inline src/main/scala/gitbucket/core/controller/AccountController.scala]: I wrote first,
```
  get("/:userName/_avatar"){
    val userName = params("userName")
    contentType = "image/png"
    getAccountByUserName(userName).map{ account =>
      response.setDateHeader("Last-Modified", account.updatedDate.getTime)
      account.image.map{ image =>
        RawData(FileUtil.getMimeType(image), new java.io.File(getUserUploadDir(userName), image))
      }.getOrElse{
        (if (account.isGroupAccount) {
          TextAvatarUtil.textGroupAvatar(account.fullName)
        } else {
          TextAvatarUtil.textAvatar(account.fullName)
        }).getOrElse(Thread.currentThread.getContextClassLoader.getResourceAsStream("noimage.png"))
      }
    }.getOrElse{
      response.setHeader("Cache-Control", "max-age=3600")
      Thread.currentThread.getContextClassLoader.getResourceAsStream("noimage.png")
    }
  }
```

It maybe redundant. `noimage.png` appears twice.
If `TextAvatarUtil` returns `None`, fallback to outer `getOrElse`.
But I didn't understand why `get` is required. Sorry for my little Scala-force.

Comment (takezoe) [inline src/main/scala/gitbucket/core/controller/AccountController.scala]: For example, maybe you can write as following in this case:

```scala
getAccountByUserName(userName).flatMap { account =>
  response.setDateHeader("Last-Modified", account.updatedDate.getTime)
  account.image.map { image =>
    Some(RawData(FileUtil.getMimeType(image), new java.io.File(getUserUploadDir(userName), image)))
  }.getOrElse {
    if (account.isGroupAccount) {
      TextAvatarUtil.textGroupAvatar(account.fullName)
    } else {
      TextAvatarUtil.textAvatar(account.fullName)
    }
  }
}.getOrElse {
  response.setHeader("Cache-Control", "max-age=3600")
  Thread.currentThread.getContextClassLoader.getResourceAsStream("noimage.png")
}
```

But I wrote this code in my brain, so I don't know whether this code works fine :-P
Could you try it?

Comment (kounoike) [inline src/main/scala/gitbucket/core/controller/AccountController.scala]: It sends empty body. :cry:

Comment (kounoike) [inline src/main/scala/gitbucket/core/controller/AccountController.scala]: Sorry, I didn't try with flatMap.

**Comment by kounoike** [comment]
@takezoe fix it by your example.
